src/compiler/parse/state/tag.ts

```diff
@@ -387,12 +387,22 @@ function read_attribute(parser: Parser, unique_names: Set<string>) {
 			}, start);
 		}
 
-		if (value[0]) {
-			if ((value as any[]).length > 1 || value[0].type === 'Text') {
-				parser.error({
-					code: 'invalid-directive-value',
-					message: 'Directive value must be a JavaScript expression enclosed in curly braces'
-				}, value[0].start);
+		const firstValue = value[0];
+		let expression = null;
+		let text = null;
+
+		if (firstValue) {
+			if ((value as any[]).length > 1 || firstValue.type === 'Text') {
+				if (type === 'Style') {
+					text = firstValue.data;
+				} else {
+					parser.error({
+						code: 'invalid-directive-value',
+						message: 'Directive value must be a JavaScript expression enclosed in curly braces'
+					}, firstValue.start);
+				}
+			} else {
+				expression = firstValue.expression || null;
```

Comment:
The Style directive is the first type of directive to allow a string literal as a value (e.g. `style:color="red"`)